src/core.js

```diff
@@ -141,26 +146,34 @@ define([
             csrftoken: _this.opts().csrftoken
         });
 
-        var hasBrokenReferences = _.debounce(function () {
-            return _this.data.core.form.hasBrokenReferences();
+        var validateForSave = _.debounce(function () {
+            var form = _this.data.core.form;
+            if (form.hasBrokenReferences()) {
+                return {
+                    title: "Errors in Form",
+                    content: "<div class='alert alert-danger'>Form has " +
+                        "reference errors.<br>Look for questions marked with " +
+                        "<i class='fd-valid-alert-icon fa fa-warning'></i> " +
+                        "and check they don't reference deleted questions.</div>",
+                };
+            } else if (!form.isFormValid(validateMug)) {
```

Comment:
I think the issue is that this happens on mouseover. I think if that's the case then we can't merge this because it will make most enterprise form building unusable.